src/Azure.AppConfiguration.Emulator.ConfigurationSettings/KeyValueProvider.cs

```diff
@@ -773,34 +769,27 @@ private async Task ScanCacheExpired(CancellationToken cancellationToken)
 
             try
             {
-                if (!_coalesceItems &&
-                    await GetExpiredItemsCount(cancellationToken) > MinCoalescingItems)
-                {
-                    _coalesceItems = true;
-                }
-
-                if (!_coalesceItems)
+                if (Interlocked.Increment(ref _coalesceItems) == 1 &&
```

Comment:
Why Increment instead of Exchange? It seems to lead to the possibility of parallel execution always having `_coalesceItems` above 1 and the scan ticks never sliding as a result.